src/test/java/com/answerdigital/benhession/academy/answerkingweek2/services/CategoryServiceTest.java

```diff
@@ -0,0 +1,303 @@
+package com.answerdigital.benhession.academy.answerkingweek2.services;
+
+import com.answerdigital.benhession.academy.answerkingweek2.exceptions.ConflictException;
+import com.answerdigital.benhession.academy.answerkingweek2.exceptions.NotFoundException;
+import com.answerdigital.benhession.academy.answerkingweek2.model.Category;
+import com.answerdigital.benhession.academy.answerkingweek2.model.Item;
+import com.answerdigital.benhession.academy.answerkingweek2.model.OrderItem;
+import com.answerdigital.benhession.academy.answerkingweek2.repositories.CategoryRepository;
+import com.answerdigital.benhession.academy.answerkingweek2.request.AddCategoryRequest;
+import com.answerdigital.benhession.academy.answerkingweek2.request.UpdateCategoryRequest;
+
+import java.math.BigDecimal;
+import java.util.HashSet;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.Optional;
+import java.util.Set;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.junit.jupiter.api.Assertions.assertSame;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.anyLong;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.any;
+import static org.mockito.Mockito.atLeast;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+class CategoryServiceTest {
+
+    private CategoryService categoryService;
+
+    @Mock
+    private CategoryRepository categoryRepository;
+
+    @Mock
+    private ItemService itemService;
+
+    Long categoryId = 1L;
+    Long itemId = 1L;
+
+    @BeforeEach
+    void setUp() {
```

Comment:
I think it would be good if we categoryService object af the end since we set it up at the beggining ?

Reply:
Good spot. I'll add a teardown method.